src/postprocess/dos.jl

```diff
@@ -68,65 +68,185 @@ function compute_ldos(scfres::NamedTuple; ε=scfres.εF, kwargs...)
+    for (iε, ε) in enumerate(εs)
+        for σ in 1:basis.model.n_spin_components, ik = krange_spin(basis, σ)
+            projsk = projections[ik]  
             @views for (iband, εnk) in enumerate(eigenvalues[ik])
-                enred = (εnk - iε) / temperature
-                D[i, :] .-= (filled_occ .* basis.kweights[ik] .* projk[iband, :]
-                             ./ temperature
-                             .* Smearing.occupation_derivative(smearing, enred))       
+                enred = (εnk - ε) / temperature
+                for iproj in 1:size(projsk, 2)
+                    D[iε, iproj, σ] -= (filled_occ * basis.kweights[ik] * projsk[iband, iproj]
+                                        / temperature
+                                        * Smearing.occupation_derivative(smearing, enred))
+                end
             end
         end
     end
-    D = mpi_sum(D, basis.comm_kpts)
+    pdos = mpi_sum(D, basis.comm_kpts)  
+
+    return (; pdos, projector_labels, εs)
 end
 
-function compute_pdos(scfres::NamedTuple, iatom, i, l; ε=scfres.εF, kwargs...)
-    psp = scfres.basis.model.atoms[iatom].psp
-    position = scfres.basis.model.positions[iatom]
-    # TODO do the symmetrization instead of unfolding    
-    scfres = unfold_bz(scfres)
-    compute_pdos(ε, scfres.basis, scfres.eigenvalues, scfres.ψ, i, l, psp, position; kwargs...)
+function compute_pdos(εs, bands; kwargs...)
+    compute_pdos(εs, bands.basis, bands.ψ, bands.eigenvalues; kwargs...)
 end
 
-# Build atomic orbitals projections projs[ik][iband,m] = |<ψnk, ϕ>|^2 for a single atom.
-# TODO optimization ? accept a range of positions, in case we want to compute the PDOS
-# for all atoms of the same type (and reuse the computation of the atomic orbitals)
-function compute_pdos_projs(basis, eigenvalues, ψ, i, l, psp::NormConservingPsp, position)
-    # Precompute the form factors on all kpoints at once so we can better use the cache (memory-compute tradeoff).
-    # Revisit this (pass the cache around explicitly) if RAM becomes an issue.
+"""
+Structure for manifold choice and projectors extraction. 
+    
+Overview of fields:
+- `iatom`: Atom position in the atoms array.
+- `species`: Chemical Element as in ElementPsp.
+- `label`: Orbital name, e.g.: "3S".
+
+All fields are optional, only the given ones will be used for selection.
+Can be called with an orbital NamedTuple and returns a boolean
+  stating whether the orbital belongs to the manifold.
+"""
+@kwdef struct OrbitalManifold
+    iatom   ::Union{Int64,  Nothing} = nothing
+    species ::Union{Symbol, AtomsBase.ChemicalSpecies, Nothing} = nothing
+    label   ::Union{String, Nothing} = nothing
+end
+function (s::OrbitalManifold)(orb)
+    iatom_match    = isnothing(s.iatom)   || (s.iatom == orb.iatom)
+    species_match  = isnothing(s.species) || (s.species == orb.species) || (orb.species == s.species) #TODO: https://github.com/JuliaMolSim/AtomsBase.jl/issues/139
```

Comment:
Space after  `#`, this should just be a comment on the line above like:
`# See https://github.com/JuliaMolSim/AtomsBase.jl/issues/139 why both species equalities are needed